### PR TITLE
updated changes

### PR DESCRIPTION
**Pull request by boniface193**
@label instance variable was created
Setter and getter for @label was added

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -39,6 +40,62 @@ def correct_name()
...
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (1GORDON) [inline person.rb]: Great job @boniface193 🤝

- [ ] As stated in the project requirements, kindly ensure a method for adding rentals should be present. You may use this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more clarity. And, the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.
add-rental book

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
 require_relative 'decorator'
 
 class Person < Namebale
```
Comment (1GORDON) [inline person.rb:4]: Great job @boniface193 🤝

- [ ] As stated in the project requirements, kindly ensure the method for adding rentals should be present. 
add-rental person

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @boniface193 ✋

Good job so far!

## Highlights!
- Created a class Student  ✔️
- Created a class person ✔️
- Created a class book ✔️
- Created a class rentals ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] I would suggest the classes created such as `book`, `rentals` etc are put in separate files.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (boniface193) [reply person.rb]: please can you put more light so that i can understand

File: person.rb (hunk 2 above)
Comment (boniface193) [reply person.rb:4]: is the comment above different from this?

File: person.rb [hunk 3]
```diff
@@ -39,6 +44,62 @@ def correct_name()
...
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (shahadat3669) [inline person.rb]: - [ ] As mentioned by the previous reviewer, you've successfully created the belongs-to side of Rental and Person (a rental belongs to a Person), which is great! However, there's also another requirement in the project to create the belongs-to side of Rental and Book(a rental belongs to a Book), and currently, this part is missing. Please consider adding the relationship function, as mentioned in the previous review.

As per the project requirement, the relationship should go both ways and currently, it doesn't. To fix this, you need to add the person to the book as you did for the book to the person. To manage both sides of the relationship in both cases, you'll need to modify your implementations of `add_rentals` in the `Book` class as well. 

This ensures that the relationship is properly established between rentals, books, and persons, meeting the project requirements. Keep up the good work! 👍📚👤

File: person.rb (hunk 2 above)
Comment (1GORDON) [reply person.rb:4]: Hello @boniface193 ✋. 
Yes, the comment is different. The references are on different classes.

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @boniface193, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Created a class Classroom
✅ Created the has-many/belongs-to relationship between Classroom and Student.
✅ Has the belongs-to side of Rental and Person (a rental belongs to a person).

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (boniface193) [reply person.rb]: Please can you explain what you mean on the second comment.

Comment (shahadat3669) [reply person.rb]: Sorry for the delayed response. I just tried to explain, there is only one requirement and you fulfilled it perfectly. 👍

Comment (boniface193) [reply person.rb]: could this be what you want me to do?

```
class Book
  attr_accessor :title, :author
  attr_reader :rentals

  def initialize(title, author)
    @title = title
    @author = author
    @rentals = []
  end

  def add_rentals(date, person)
    Rental.new(date, self, person)
  end
end
```

**Review by shahadat3669 — APPROVED** [review]
Hi @boniface193, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Created the has-many/belongs-to relationship between Classroom and Student.
✅ Has the belongs-to side of Rental and Person (a rental belongs to a person).
✅ Has the belongs-to side of Rental and Book (a rental belongs to a Book). 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**